File: app/src/main/java/me/zoe/bot/commons/utils/CommandContext.java
package me.zoe.bot.commons.utils;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.entities.*;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
import net.dv8tion.jda.api.exceptions.ErrorResponseException;

import me.zoe.bot.Zoe;

import java.util.concurrent.TimeUnit;

public class CommandContext {

    private final Zoe bot;
    private final GuildMessageReceivedEvent event;

    public CommandContext(Zoe bot, GuildMessageReceivedEvent event) {
        this.bot = bot;
        this.event = event;
    }

    public GuildMessageReceivedEvent getEvent() {
        return event;
    }

    public void reply(String message) {
        event.getChannel().sendMessage(message).queue();
    }

    public void reply(EmbedBuilder embedBuilder) {
        event.getChannel().sendMessage(embedBuilder.build()).queue();
    }

    public void reply(MessageEmbed messageEmbed) {
        event.getChannel().sendMessage(messageEmbed).queue();
    }

    public void timedReply(String message, int time, TimeUnit timeUnit) {
        event.getChannel().sendMessage(message).queue(msg -> msg.delete().queueAfter(time, timeUnit));
    }

    public void timedReply(EmbedBuilder embedBuilder, int time, TimeUnit timeUnit) {
        event.getChannel().sendMessage(embedBuilder.build()).queue(msg -> msg.delete().queueAfter(time, timeUnit));
    }

    public void timedReply(MessageEmbed messageEmbed, int time, TimeUnit timeUnit) {
        event.getChannel().sendMessage(messageEmbed).queue(msg -> msg.delete().queueAfter(time, timeUnit));
    }

    public void deleteCmd() {
        event.getMessage().delete().queue();
    }

    public void deleteCmd(int time, TimeUnit timeUnit) {
        try {
            event.getMessage().delete().queueAfter(time, timeUnit);
        } catch (ErrorResponseException ignored) {
        }
    }

    public Member getSelfMember() {
        return event.getGuild().getSelfMember();
    }

    public void kickMember(User user) {
        event.getGuild().getMember(user).kick().queue();
    }

    public void kickMember(User user, String reason) {
        event.getGuild().getMember(user).kick(reason).queue();
    }

    public void banMember(User user, int days) {
        event.getGuild().getMember(user).ban(days).queue();
    }

    public void banMember(User user, int days, String reason) {
        event.getGuild().getMember(user).ban(days, reason).queue();
    }

    public void banMember(Guild guild, User user, int days) {
        guild.getMember(user).ban(days).queue();
    }

    public void banMember(Guild guild, User user, int days, String reason) {
        guild.getMember(user).ban(days, reason).queue();
    }

    public User getAuthor() {
        return event.getAuthor();
    }

    public MessageChannel getChannel() {
        return event.getChannel();
    }

    public Guild getGuild() {
        return event.getGuild();
    }

    public JDA getJDA() {
        return event.getJDA();
    }

    public Member getMember() {
        return event.getMember();
    }

    public Message getMessage() {
        return event.getMessage();
    }

    public PrivateChannel getPrivateChannel() {
        return event.getAuthor().openPrivateChannel().complete();
    }

    public long getResponseNumber() {
        return event.getResponseNumber();
    }
}

File: src/main/kotlin/com/github/mrdrooxbr/rioru/bot/commands/framework/CommandContext.kt
package com.github.mrdrooxbr.rioru.bot.commands.framework

import com.github.mrdrooxbr.rioru.bot.commands.framework.rioru.RioruEmojis
import com.github.mrdrooxbr.rioru.bot.commands.framework.rioru.RioruTranslates
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
import net.dv8tion.jda.api.sharding.ShardManager
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.entities.*

class CommandContext(private var event: GuildMessageReceivedEvent, private var args: List<String>, private val locale: String) {
    private val translates = RioruTranslates()
    fun getGuild(): Guild = event.guild
    fun getArgs(): List<String> = args
    fun getChannel(): TextChannel = event.channel
    fun getMessage(): Message = event.message
    fun getAuthor(): User = event.author
    fun getMember(): Member? = event.member
    fun getJDA(): JDA = event.jda
    fun getShardManager(): ShardManager? = getJDA().shardManager
    fun getSelfUser(): User = getJDA().selfUser
    fun getSelfMember(): Member = getGuild().selfMember
    fun sendMessage(embed: EmbedBuilder) = getChannel().sendMessage(embed.build()).queue()
    fun translate(emoji: String, translateUri: String): String {
        val emote = RioruEmojis.get(emoji)
        val uri = translateUri.split("|")
        return if(uri.size === 2) {
            "$emote | " + translates.get(locale, uri[0]).getString(uri[1])
        } else {
            "$emote | " + translates.get(locale, uri[0]).getJSONObject(uri[1]).getString(uri[2])
        }
    }

    fun translate(translateUri: String): String {
        val uri = translateUri.split("|")
        return if(uri.size === 2) {
            translates.get(locale, uri[0]).getString(uri[1])
        } else {
            translates.get(locale, uri[0]).getJSONObject(uri[1]).getString(uri[2])
        }
    }

}
